src/nbl/video/IPhysicalDevice.cpp

```diff
@@ -103,58 +103,226 @@ inline core::bitflag<asset::IImage::E_ASPECT_FLAGS> getImageAspects(asset::E_FOR
+
+        // Otherwise, assume binary (1 bit) alpha
+        default:
+            return 1.f;
+        }
+    }
+
+    float rcpUnit = 0.0;
+    switch (format)
+    {
+    case asset::EF_BC1_RGB_UNORM_BLOCK:
+    case asset::EF_BC1_RGB_SRGB_BLOCK:
+    case asset::EF_BC1_RGBA_UNORM_BLOCK:
+    case asset::EF_BC1_RGBA_SRGB_BLOCK:
+    case asset::EF_BC2_UNORM_BLOCK:
+    case asset::EF_BC2_SRGB_BLOCK:
+    case asset::EF_BC3_UNORM_BLOCK:
+    case asset::EF_BC3_SRGB_BLOCK:
+        rcpUnit = (channel == 1u) ? (1.0 / 63.0) : (1.0 / 31.0);
+        break;
+    case asset::EF_BC4_UNORM_BLOCK:
+    case asset::EF_BC4_SNORM_BLOCK:
+    case asset::EF_BC5_UNORM_BLOCK:
+    case asset::EF_BC5_SNORM_BLOCK:
+    case asset::EF_BC7_UNORM_BLOCK:
+    case asset::EF_BC7_SRGB_BLOCK:
+        rcpUnit = 1.0 / 255.0;
+        break;
+    case asset::EF_ASTC_4x4_UNORM_BLOCK:
+    case asset::EF_ASTC_4x4_SRGB_BLOCK:
+    case asset::EF_ASTC_5x4_UNORM_BLOCK:
+    case asset::EF_ASTC_5x4_SRGB_BLOCK:
+    case asset::EF_ASTC_5x5_UNORM_BLOCK:
+    case asset::EF_ASTC_5x5_SRGB_BLOCK:
+    case asset::EF_ASTC_6x5_UNORM_BLOCK:
+    case asset::EF_ASTC_6x5_SRGB_BLOCK:
+    case asset::EF_ASTC_6x6_UNORM_BLOCK:
+    case asset::EF_ASTC_6x6_SRGB_BLOCK:
+    case asset::EF_ASTC_8x5_UNORM_BLOCK:
+    case asset::EF_ASTC_8x5_SRGB_BLOCK:
+    case asset::EF_ASTC_8x6_UNORM_BLOCK:
+    case asset::EF_ASTC_8x6_SRGB_BLOCK:
+    case asset::EF_ASTC_8x8_UNORM_BLOCK:
+    case asset::EF_ASTC_8x8_SRGB_BLOCK:
+    case asset::EF_ASTC_10x5_UNORM_BLOCK:
+    case asset::EF_ASTC_10x5_SRGB_BLOCK:
+    case asset::EF_ASTC_10x6_UNORM_BLOCK:
+    case asset::EF_ASTC_10x6_SRGB_BLOCK:
+    case asset::EF_ASTC_10x8_UNORM_BLOCK:
+    case asset::EF_ASTC_10x8_SRGB_BLOCK:
+    case asset::EF_ASTC_10x10_UNORM_BLOCK:
+    case asset::EF_ASTC_10x10_SRGB_BLOCK:
+    case asset::EF_ASTC_12x10_UNORM_BLOCK:
+    case asset::EF_ASTC_12x10_SRGB_BLOCK:
+    case asset::EF_ASTC_12x12_UNORM_BLOCK:
+    case asset::EF_ASTC_12x12_SRGB_BLOCK:
+        // (All of these could be using HDR. Take extra flag to assume FP16 precision?)
+        rcpUnit = 1.0 / 255.0;
+        break;
+    case asset::EF_EAC_R11_UNORM_BLOCK:
+    case asset::EF_EAC_R11_SNORM_BLOCK:
+    case asset::EF_EAC_R11G11_UNORM_BLOCK:
+    case asset::EF_EAC_R11G11_SNORM_BLOCK:
+        rcpUnit = 1.0 / 2047.0;
+        break;
+    case asset::EF_ETC2_R8G8B8_UNORM_BLOCK:
+    case asset::EF_ETC2_R8G8B8_SRGB_BLOCK:
+    case asset::EF_ETC2_R8G8B8A1_UNORM_BLOCK:
+    case asset::EF_ETC2_R8G8B8A1_SRGB_BLOCK:
+    case asset::EF_ETC2_R8G8B8A8_UNORM_BLOCK:
+    case asset::EF_ETC2_R8G8B8A8_SRGB_BLOCK:
+        rcpUnit = 1.0 / 31.0;
+        break;
+    case asset::EF_BC6H_UFLOAT_BLOCK:
+    case asset::EF_BC6H_SFLOAT_BLOCK:
+    {
+        // BC6 isn't really FP16, so this is an over-estimation
+        float f = 0.0;
+        uint16_t f16 = core::Float16Compressor::compress(f);
+        uint16_t dir = core::Float16Compressor::compress(2.f * (f + 1.f));
```

Comment:
IEEE754 **positive** floats are ordered same way as their bitrep, you can use `0x0001u`, unless you want to use minimum normalized value to account for "denormalized flush to 0 on GPUs"